php-checks/src/main/java/org/sonar/php/checks/OverridingMethodSimplyCallParentCheck.java

```diff
@@ -19,116 +19,115 @@
-        String expectedExpression = "(?:parent|" + parentClassName.replace("\\", "\\\\") + ")::" + methodName + "\\(" + expectedParam + "\\)";
+      if (blockTree.statements().size() == 1) {
+        StatementTree statementTree = blockTree.statements().get(0);
 
-        return Pattern.matches(expectedExpression, expression);
-      }
-    }
-    return false;
-  }
+        ExpressionTree expressionTree = null;
+        if (statementTree.is(Kind.EXPRESSION_STATEMENT)) {
+          expressionTree = ((ExpressionStatementTree) statementTree).expression();
 
-  private String buildExpectedParamRegexp(int nbParam) {
-    String expectedParam;
+        } else if (statementTree.is(Kind.RETURN_STATEMENT)) {
+          expressionTree = ((ReturnStatementTree) statementTree).expression();
+        }
 
-    if (nbParam == 0) {
-      expectedParam = "";
-    } else if (nbParam == 1) {
-      expectedParam = LexicalConstant.VAR_IDENTIFIER;
-    } else {
-      expectedParam = "(?:" + LexicalConstant.VAR_IDENTIFIER + ",){" + (nbParam - 1) + ",}" + LexicalConstant.VAR_IDENTIFIER;
+        checkExpression(expressionTree, method);
+      }
     }
-    return expectedParam;
   }
 
-  private String getExpressionAsString(AstNode stmtList) {
-    AstNode stmt = stmtList.getFirstChild().getFirstChild();
+  private void checkExpression(@Nullable ExpressionTree expressionTree, MethodDeclarationTree method) {
+    if (expressionTree != null && expressionTree.is(Kind.FUNCTION_CALL)) {
+      FunctionCallTree functionCallTree = (FunctionCallTree)expressionTree;
 
-    if (stmt.is(PHPGrammar.RETURN_STATEMENT) && stmt.hasDirectChildren(PHPGrammar.EXPRESSION)) {
-      return CheckUtils.getExpressionAsString(stmt.getFirstChild(PHPGrammar.EXPRESSION));
+      if (functionCallTree.callee().is(Kind.CLASS_MEMBER_ACCESS)) {
+        MemberAccessTree memberAccessTree = (MemberAccessTree)functionCallTree.callee();
 
-    } else if (stmt.is(PHPGrammar.EXPRESSION_STATEMENT)) {
-      return CheckUtils.getExpressionAsString(stmt.getFirstChild(PHPGrammar.EXPRESSION));
+        String methodName = method.name().text();
+        boolean sameMethodName = CheckUtils.asString(memberAccessTree.member()).equals(methodName);
 
-    } else {
-      return null;
+        if (isSuperClass(memberAccessTree.object()) && sameMethodName && sameArguments(functionCallTree, method)) {
+          String message = String.format(MESSAGE, methodName);
+          context().newIssue(KEY, message).tree(method);
+        }
+      }
     }
   }
 
-  private int getNumberOfParameter(AstNode paramList) {
-    return paramList != null ? paramList.getChildren(PHPGrammar.PARAMETER).size() : 0;
+  private boolean sameArguments(FunctionCallTree functionCallTree, MethodDeclarationTree method) {
+    boolean sameArgumentsNumber = functionCallTree.arguments().size() == method.parameters().parameters().size();
+    return sameArgumentsNumber && allArgumentsAreTrivial(functionCallTree);
```

Comment:
@vilchik-elena checking only that arguments are simple variables may lead to false positives, e.g.
`public f($p1, $p2) {
   parent::f($p2, $p1);
}`